recipes/mpg123/all/conanfile.py

```diff
@@ -43,153 +50,162 @@ class Mpg123Conan(ConanFile):
+            yes_no = lambda v: "yes" if v else "no"
+            tc = AutotoolsToolchain(self)
+            tc.configure_args.extend([
+                f"--enable-moreinfo={yes_no(self.options.moreinfo)}",
+                f"--enable-network={yes_no(self.options.network)}",
+                f"--enable-ntom={yes_no(self.options.flexible_resampling)}",
+                f"--enable-icy={yes_no(self.options.icy)}",
+                f"--enable-id3v2={yes_no(self.options.id3v2)}",
+                f"--enable-ieeefloat={yes_no(self.options.ieeefloat)}",
+                f"--enable-layer1={yes_no(self.options.layer1)}",
+                f"--enable-layer2={yes_no(self.options.layer2)}",
+                f"--enable-layer3={yes_no(self.options.layer3)}",
+                f"--with-audio={self._audio_module}",
+                f"--with-default-audio={self._audio_module}",
+                f"--with-seektable={self.options.seektable}",
+                f"--enable-modules=no",
+                f"--enable-shared={yes_no(self.options.shared)}",
+                f"--enable-static={yes_no(not self.options.shared)}",
+            ])
+            tc.generate()
+            tc = AutotoolsDeps(self)
+            tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        if self.settings.compiler == "Visual Studio":
-            cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        if is_msvc(self):
+            cmake = CMake(self)
+            cmake.configure(build_script_folder=os.path.join(self.source_folder, "ports", "cmake"))
             cmake.build()
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
+            autotools.configure()
             autotools.make()
 
     def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        if self.settings.compiler == "Visual Studio":
-            cmake = self._configure_cmake()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if is_msvc(self):
+            cmake = CMake(self)
             cmake.install()
-            tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
         else:
-            autotools = self._configure_autotools()
+            autotools = Autotools(self)
             autotools.install()
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+            rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
```

Comment:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
        
        fix_apple_shared_install_name(self)
```

when `shared=True` the libraries had absolute path in their install name :D